components/org.wso2.carbon.identity.local.auth.push.authenticator/src/main/java/org/wso2/carbon/identity/local/auth/push/authenticator/PushAuthenticator.java

```diff
@@ -0,0 +1,1702 @@
+                        return;
+                    }
+                }
+            }
+
+            Device device;
+            try {
+                device = AuthenticatorDataHolder.getInstance().getDeviceHandlerService()
+                        .getDeviceByUserId(authenticatedUserFromContext.getUserId(), tenantDomain);
+            } catch (UserIdNotFoundException e) {
+                String error = String.format(
+                        ERROR_CODE_ERROR_GETTING_USER_ID.getMessage(),
+                        AuthenticatorUtils.maskIfRequired(authenticatedUserFromContext.getUserName()));
+                throw new AuthenticationFailedException(ERROR_CODE_ERROR_GETTING_USER_ID.getCode(),
+                        error, e);
+            } catch (PushDeviceHandlerException e) {
+                if (ERROR_CODE_DEVICE_NOT_FOUND_FOR_USER_ID.getCode().equals(e.getErrorCode())) {
+                    // The user does not have a device registered.
+                    device = null;
+                } else {
+                    String error = String.format(
+                            ERROR_CODE_ERROR_GETTING_USER_DEVICE.getMessage(),
+                            AuthenticatorUtils.maskIfRequired(authenticatedUserFromContext.getUserName()));
+                    throw new AuthenticationFailedException(ERROR_CODE_ERROR_GETTING_USER_DEVICE.getCode(),
+                            error, e);
+                }
+            }
+
+            // If device is null, that means the user does not have a device registered.
+            if (device == null) {
+
+                // Check if push device progressive enrollment is enabled.
+                if (!isProgressiveDeviceEnrollmentEnabled(tenantDomain)) {
+                    handlePushAuthFailedScenario(request, response, context, ERROR_USER_REGISTERED_DEVICE_NOT_FOUND);
+                    return;
+                }
+
+                if (isUserRedirectedFromIDF(request)) {
+
+                    if (context.getProperty(IS_DEVICE_REGISTRATION_CONSENT_GIVEN) != null && Boolean.TRUE.equals(
+                            context.getProperty(IS_DEVICE_REGISTRATION_CONSENT_GIVEN))) {
+                        context.removeProperty(IS_DEVICE_REGISTRATION_CONSENT_GIVEN);
+                    } else {
+                        // If there is no device registered for the user, and the user is redirected from the Identifier
+                        // then, get consent from the user to register a device.
+                        handleIDFUserDeviceEnrolConsentScenario(
+                                authenticatedUserFromContext, response, request, context);
+                        if (!StringUtils.isEmpty(request.getParameter(USERNAME))) {
+                            persistUsername(context, request.getParameter(USERNAME));
+                        }
+                        context.setProperty(IS_DEVICE_REGISTRATION_CONSENT_GIVEN, true);
```

Comment:
```suggestion
                        context.setProperty(IS_DEVICE_REGISTRATION_ENGAGED, true);
```
Lets change the wording as we use the term consent with a different meaning than it is used in here